### DISCUSSION TITLE
setData 和 replaceData 删除 数组数据 不能触发 node.on(change:data,()=>{}) 函数 ！！！！！

### DISCUSSION THREAD

**Discussion by lance-li** [antvis/X6#2254] [Q&A]
<img width="1088" alt="image" src="https://user-images.githubusercontent.com/21050392/175024841-13756b67-2d4a-40f1-9108-a5687b5885e5.png">

**Comment by journey0726** [comment]
我也遇到这个问题了

**Comment by FatOcean** [comment]
这个问题解决了吗

**Comment by Miroxyz** [comment]
3202年了，这个问题好像还没解决

**Comment by maqingbo** [comment]
可能是个bug，需要先调用 removeData，然后在调用 setData。参考 #3262

**Comment by maqingbo** [comment]
可能是个bug，需要先调用 removeData，然后在调用 setData。参考 #3262
